streams/src/test/java/org/apache/kafka/streams/integration/ResetIntegrationTest.java

```diff
@@ -374,25 +385,15 @@ private void assertInternalTopicsGotDeleted(final String intermediateUserTopic)
         expectedRemainingTopicsAfterCleanup.add(OUTPUT_TOPIC_2_RERUN);
         expectedRemainingTopicsAfterCleanup.add("__consumer_offsets");
 
-        Set<String> allTopics;
-        ZkUtils zkUtils = null;
+        Set<String> allTopics = new HashSet<>();
         try {
-            zkUtils = ZkUtils.apply(CLUSTER.zKConnectString(),
-                30000,
-                30000,
-                JaasUtils.isZkSecurityEnabled());
-
-            do {
-                Utils.sleep(100);
-                allTopics = new HashSet<>();
-                allTopics.addAll(scala.collection.JavaConversions.seqAsJavaList(zkUtils.getAllTopics()));
-            } while (allTopics.size() != expectedRemainingTopicsAfterCleanup.size());
-        } finally {
-            if (zkUtils != null) {
-                zkUtils.close();
-            }
+            ListTopicsOptions listTopicsOptions = new ListTopicsOptions();
+            listTopicsOptions.listInternal(true);
+            allTopics.addAll(kafkaAdminClient.listTopics(listTopicsOptions).names().get(30000, TimeUnit.MILLISECONDS));
+            assertThat(allTopics, equalTo(expectedRemainingTopicsAfterCleanup));
+        } catch (Exception e) {
```

Comment:
final

Why do we need to catch an fail here? We can just let the exception bubble out to fail the test

Reply:
I'd prefer to leave as is because IMHO the `Assert.fail` is more immediately clear as to why the test failed.

Reply:
Not sure what @dguy and @guozhangwang think. From my understanding, one should never catch an exception in a test (with the exception you want to do a negative test, ie, the exception implies that the test passes).

Reply:
Fair enough, I'll take it out.